### PR TITLE
RHPAM-2532: removing git write access and disabling by default the git protocol

### PR DESCRIPTION
**Pull request by porcelli**
Requires the merge of the following PRs too:
- https://github.com/kiegroup/kie-wb-distributions/pull/989

### CODE REVIEW

**Comment by porcelli** [comment]
jenkins execute full downstream build

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemImplProviderSSHTest.java [hunk 1]
```diff
@@ -19,42 +19,50 @@
...
+    private int gitPort;
 
     @Override
     public Map<String, String> getGitPreferences() {
-        Map<String, String> gitPrefs = super.getGitPreferences();
-
-        gitPrefs.put("org.uberfire.nio.git.ssh.enabled",
-                     "true");
+        final Map<String, String> gitPrefs = super.getGitPreferences();
+        gitPrefs.put("org.uberfire.nio.git.daemon.enabled", "true");
```
Comment (tomasdavidorg) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemImplProviderSSHTest.java]:
```suggestion
        gitPrefs.put(GIT_DAEMON_ENABLED, "true");
```

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemImplProviderSSHTest.java [hunk 2]
```diff
@@ -19,42 +19,50 @@
...
+        gitPort = findFreePort();
+        gitPrefs.put(GIT_DAEMON_PORT, String.valueOf(gitPort));
+        gitPrefs.put("org.uberfire.nio.git.ssh.enabled", "true");
```
Comment (tomasdavidorg) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemImplProviderSSHTest.java]:
```suggestion
        gitPrefs.put(GIT_SSH_ENABLED, "true");
```

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemImplProviderSSHTest.java [hunk 3]
```diff
@@ -19,42 +19,50 @@
...
         gitSSHPort = findFreePort();
-        gitPrefs.put("org.uberfire.nio.git.ssh.port",
-                     String.valueOf(gitSSHPort));
-        gitPrefs.put("org.uberfire.nio.git.ssh.idle.timeout",
-                     "10001");
+        gitPrefs.put("org.uberfire.nio.git.ssh.port", String.valueOf(gitSSHPort));
```
Comment (tomasdavidorg) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemImplProviderSSHTest.java]:
```suggestion
        gitPrefs.put(GIT_SSH_PORT, String.valueOf(gitSSHPort));
```

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemImplProviderSSHTest.java [hunk 4]
```diff
@@ -19,42 +19,50 @@
...
+        gitPrefs.put("org.uberfire.nio.git.ssh.idle.timeout", "10001");
```
Comment (tomasdavidorg) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemImplProviderSSHTest.java]:
```suggestion
        gitPrefs.put(GIT_SSH_IDLE_TIMEOUT, "10001");
```

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemImplProviderSSHTest.java [hunk 5]
```diff
@@ -19,42 +19,50 @@
...
 import org.uberfire.java.nio.file.extensions.FileSystemHooks;
 import org.uberfire.java.nio.fs.jgit.util.commands.Commit;
+import org.uberfire.java.nio.fs.jgit.util.commands.Push;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
 import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.verify;
+import static org.uberfire.java.nio.fs.jgit.JGitFileSystemProviderConfiguration.GIT_DAEMON_PORT;
```
Comment (tomasdavidorg) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemImplProviderSSHTest.java]:
```suggestion
import static org.uberfire.java.nio.fs.jgit.JGitFileSystemProviderConfiguration.GIT_DAEMON_ENABLED;
import static org.uberfire.java.nio.fs.jgit.JGitFileSystemProviderConfiguration.GIT_DAEMON_PORT;
import static org.uberfire.java.nio.fs.jgit.JGitFileSystemProviderConfiguration.GIT_SSH_ENABLED;
import static org.uberfire.java.nio.fs.jgit.JGitFileSystemProviderConfiguration.GIT_SSH_IDLE_TIMEOUT;
import static org.uberfire.java.nio.fs.jgit.JGitFileSystemProviderConfiguration.GIT_SSH_PORT;
```

**Comment by porcelli** [comment]
jenkins execute full downstream build

**Comment by paulovmr** [comment]
The test failure was fixed and passed in the related PR build.

**Pull request merged**